KarhooUISDK/CommonUI/BottomSheet/KarhooBottomSheet.swift

```diff
@@ -23,104 +23,93 @@ struct KarhooBottomSheet<Content: View>: View {
    
     // MARK: - Body
     var body: some View {
-        VStack(spacing: 0.0, content: {
+        ZStack {
             Spacer()
-                .background(Color.clear)
-                .layoutPriority(Double(UILayoutPriority.defaultLow.rawValue))
+                .background(
+                    LinearGradient(
+                        colors: [
+                            Color(KarhooUI.colors.black).opacity(UIConstants.Alpha.hidden),
+                            Color(KarhooUI.colors.black).opacity(UIConstants.Alpha.enabled)
+                        ],
+                        startPoint: .top,
+                        endPoint: .bottom
+                    )
+                )
+                .edgesIgnoringSafeArea([.all])
+                .opacity(2 - Double(offset.height / 100))
+                .gesture(
+                    DragGesture()
+                        .onChanged({ gesture in
+                            if gesture.translation.height > 0 {
+                                offset = gesture.translation
+                            }
+                        })
+                        .onEnded({ _ in
+                            if offset.height > 100 {
+                                viewModel.dismiss()
+                            } else {
+                                offset = .zero
+                            }
+                        })
+                )
+                .onTapGesture {
+                    viewModel.dismiss()
+                }
             
-            VStack(spacing: 0.0) {
-                HStack {
-                    Text(viewModel.title)
-                        .font(Font(KarhooUI.fonts.title2Bold()))
-                        .foregroundColor(Color(KarhooUI.colors.text))
-                    Spacer()
-                        .background(Color.clear)
-                    Button {
-                        viewModel.dismiss()
-                    } label: {
-                        Image(
-                            uiImage:
-                                UIImage.uisdkImage("kh_uisdk_cross_new")
-                                .coloured(withTint: KarhooUI.colors.text)
+            VStack(spacing: 0) {
+                Spacer()
+                    .background(Color.clear)
+                VStack(spacing: 0) {
+                    HStack {
+                        Text(viewModel.title)
+                            .font(Font(KarhooUI.fonts.title2Bold()))
+                            .foregroundColor(Color(KarhooUI.colors.text))
+                        Spacer()
+                            .background(Color.clear)
+                        Button {
```

Comment:
Tbh I personally prefer using full invocation of `Button` view - feel it more organized, than using two colures.

```
Button(
    action: {
        viewModel.dismisss()
    },
    label: { 
        Image()...
    }
}
```

It's not an request, only suggestion :D